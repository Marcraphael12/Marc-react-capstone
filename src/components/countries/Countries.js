import React, { useEffect, useState } from 'react';
import PropTypes from 'prop-types';
import Country from './country';
import Query from '../Search';
import './scss/countries.scss';

const Countries = ({
  countries, total, loading, handleClick,
}) => {
  const [filtered, setFiltered] = useState(Object.keys(countries));

  const filterCountries = (expression) => Object.keys(countries).filter((country) => {
    const regex = new RegExp(expression, 'gi');
    return country.match(regex);
  });

  const handleChange = (event) => setFiltered(filterCountries(event.target.value));

  useEffect(() => {
    setFiltered(Object.keys(countries));
  }, [countries]);

  return (
    <div className="header-container">
      <div className="header-wrapper">
        <div className="world-info">
          <span className="world-cases">Roughly </span>
          <span className="world-cases-value">
            {' '}
            {total && Intl.NumberFormat('de-DE').format(total.today_confirmed)}
          </span>
          <span className="world-cases">Covid world Cases </span>
        </div>
        <Query handleChange={handleChange} />
      </div>

      <ul className="countries-list">
        {filtered && filtered.map((key) => (
          (key !== 'pistolet')
          && (
            <li key={key} className="countries">
              <Country
                country={key}
                totalConfirmed={countries[key].today_confirmed}
                handleClick={handleClick}
              />
            </li>
          )
        ))}
      </ul>
      {!filterCountries.length && !loading && (
        <span>
          Sorry, No match available !
          {' '}
          {' '}
        </span>
      )}
    </div>
  );
};

Countries.defaultProps = {
  loading: true,
  total: {},
};

Countries.propTypes = {
  countries: PropTypes.objectOf(PropTypes.object).isRequired,
  total: PropTypes.shape({
    today_confirmed: PropTypes.number,
  }),
  handleClick: PropTypes.func.isRequired,
  loading: PropTypes.bool,

};

export default Countries;