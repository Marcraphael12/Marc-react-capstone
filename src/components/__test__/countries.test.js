import React from 'react';
import renderer from 'react-test-renderer';
import Countries from '../countries/Countries';
import { HashRouter } from 'react-router-dom';

const mockedData = {
  Benin: {
    new_cases: 4,
    death: 0,
    today_confirmed: 24850,
  },
};

const mockedTotalCases = {
  today_confirmed: 33868,
  today_new_confirmed: 6753375,
  today_deaths: 575238,
};

describe('Countries', () => {
  it('renders ! ', () => {
    const tree = renderer
      .create(
        <HashRouter>
        <Countries
          countries={mockedData}
          total={mockedTotalCases}
          loading={false}
          handleClick={() => 7 + 5}
        />
        </HashRouter>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});