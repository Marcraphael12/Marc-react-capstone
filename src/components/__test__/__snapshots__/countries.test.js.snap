// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Countries renders !  1`] = `
<div
  className="header-container"
>
  <div
    className="header-wrapper"
  >
    <div
      className="world-info"
    >
      <span
        className="world-cases"
      >
        Roughly 
      </span>
      <span
        className="world-cases-value"
      >
         
        33.868
      </span>
      <span
        className="world-cases"
      >
        Covid world Cases 
      </span>
    </div>
    <form
      className="form"
    >
      <input
        className="input-search"
        onChange={[Function]}
        placeholder="Search a country"
        type="text"
      />
    </form>
  </div>
  <ul
    className="countries-list"
  >
    <li
      className="countries"
    >
      <button
        className="country"
        onClick={[Function]}
        type="button"
      >
        <svg
          className="info-icon"
          color="white"
          fill="currentColor"
          height={20}
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": "white",
            }
          }
          viewBox="0 0 24 24"
          width={20}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12 2C6.486 2 2 6.486 2 12s4.486 10 10 10 10-4.486 10-10S17.514 2 12 2zm0 18c-4.411 0-8-3.589-8-8s3.589-8 8-8 8 3.589 8 8-3.589 8-8 8z"
          />
          <path
            d="M9.293 7.707 13.586 12l-4.293 4.293 1.414 1.414L16.414 12l-5.707-5.707z"
          />
        </svg>
        <div
          className="country-data"
        >
          <h1
            className="country-name"
          >
            BENIN
          </h1>
          <span
            className="total-confirmed"
          >
            24.850
          </span>
        </div>
      </button>
    </li>
  </ul>
</div>
`;
